lib/matplotlib/testing/widgets.py

```diff
@@ -18,31 +18,35 @@ def get_ax():
     return ax
 
 
-def do_event(tool, etype, button=1, xdata=0, ydata=0, key=None, step=1):
-    """
-    Trigger an event
+def mock_event(ax, button=1, xdata=0, ydata=0, key=None, step=1):
```

Comment:
Git diff looks more complicated than it is. This just extracts the event creation out of `do_event()`.